### PR TITLE
Bug 1784232 - Let CoT support BASE_REF and BASE_REV

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Depends on https://phabricator.services.mozilla.com/D155083

### CODE REVIEW

**Review by escapewindow — APPROVED** [review]
I was concerned with backwards compatibility, but all the cot tests passed. Thanks!

File: src/scriptworker/constants.py [hunk 1]
```diff
@@ -110,7 +110,7 @@
             }
         ),
         "project_configuration_url": "https://hg.mozilla.org/ci/ci-configuration/raw-file/default/projects.yml",
-        "pushlog_url": "{repo}/json-pushes?changeset={revision}&tipsonly=1&version=2&full=1",
+        "pushlog_url": "{repo}/json-pushes?changeset={revision}&version=2&full=1",
```
Comment (jcristau) [inline src/scriptworker/constants.py:113]: I'm curious about this, I couldn't find where we need the other revisions.

File: src/scriptworker/cot/verify.py [hunk 2]
```diff
@@ -1037,8 +1039,16 @@ async def _get_additional_hg_action_jsone_context(parent_link, decision_link):
 async def _get_additional_hgpush_info(decision_link):
     pushlog_info = await get_pushlog_info(decision_link)
     pushlog_id = list(pushlog_info["pushes"].keys())[0]
-    push_comment = pushlog_info["pushes"][pushlog_id]["changesets"][0]["desc"]
-    return {"id": pushlog_id, "comment": push_comment, "date": pushlog_info["pushes"][pushlog_id]["date"], "user": pushlog_info["pushes"][pushlog_id]["user"]}
+    first_revision = pushlog_info["pushes"][pushlog_id]["changesets"][0]
+    push_comment = first_revision["desc"]
```
Comment (jcristau) [inline src/scriptworker/cot/verify.py:1043]: Shouldn't this be the tip commit's description, to preserve the previous behavior?

Comment (JohanLorenzo) [reply src/scriptworker/cot/verify.py]: Oh yeah, definitely! Thanks for pointing that out. Fixed!

**Pull request merged**